.github/workflows/test.yml

```diff
@@ -37,6 +37,7 @@ jobs:
     runs-on: ubuntu-latest
     # Only run for a tagged commit
     if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+    environment: pypi
```

Comment:
The line you have added is optional. 

I think the essential one is `permissions: id-token: write` based on https://docs.pypi.org/trusted-publishers/using-a-publisher/

Reply:
I removed the line. I think we can go ahead and test if the publishing now works

Reply:
I think you also need to add the one that is they say is essential